hack/codegen/prototag/main.go

```diff
@@ -0,0 +1,86 @@
+package main
```

Comment:
Can we get some comments in here to document what this tool is used for?

Reply:
Add comments: https://github.com/akuity/kargo/pull/1515/commits/003b4607215101c2fd9c70b2c7d8c9bcb096f0d2